### PR TITLE
Adds Key Vault name

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/azure-operator#7

This PR adds the Azure Key Vault name to the TPR. The name needs to be globally unique and the operator needs to know it for the custom unit to download secrets and in future to update those secrets. So I think it needs to be part of the cluster custom object.

> Key Vault names are selected by the user and are globally unique.
> Key Vault name must be a string 3-24 characters in length containing only (0-9, a-z, A-Z, and -).

https://docs.microsoft.com/en-us/rest/api/keyvault/about-keys--secrets-and-certificates

### CODE REVIEW

File: spec/azure/keyvault.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package azure
+
+type KeyVault struct {
+	// VaultName is the Azure Key Vault name.
+	VaultName string `json:"vaultName" yaml:"vaultName"`
```
Comment (kopiczko) [inline spec/azure/keyvault.go]: 1. Maybe just `Name`?
2. Please put some good stuff form PR description into the godoc.

**Review by kopiczko — APPROVED** [review]
Small requests. LGTM.

File: spec/azure/keyvault.go (hunk 1 above)
Comment (rossf7) [reply spec/azure/keyvault.go]: Yes I think `Name` is better. I'll change and add the extra info.

Comment (rossf7) [reply spec/azure/keyvault.go]: Done

**Pull request merged**